### PR TITLE
feat: misp event reports

### LINKED ISSUE #130: `misp_event_reports` publisher in `misp` plugin
### Description
MISP allows users to attach analysis to any event. [Here](https://www.misp-project.org/2020/10/08/Event-Reports.html/) is a blog post on how to use it in UI.

### Use Case
It would be very useful to be able to push the produced report straight into the MISP instance and attach it to the event.

### Requirements

- configuration attributes
  - `api_key` - (required) string attribute
- execution attributes (taken from [Add Event Report endpoint docs](https://www.misp-project.org/openapi/#tag/Event-Report/operation/addEventReport))
  - `event_id` - (required) string attribute
  - `name` - (required) string attribute
  - `distribution` - (optional) int attribute, enum
  - `sharing_group_id` - (optional) string attribute

### Behaviour
- `name` must support Go template strings, so that the name can be rendered from data in the context
- `uuid` field in the request - Fabric should generate a unique UUID value
- `content` field in the request should contain a rendered document in the format specified in [`doc_format` generic attribute](https://github.com/blackstork-io/fabric/issues/126)
  - the data sink supports only markdown output format
- `deleted` parameter in the request is always `false`
- `timestamp` is filled in by Fabric
- after the push, the data sink should write an info log message with `EventReport.id`, `EventReport.uuid`, `EventReport.event_id`, `EventReport.name` values from the response received from the endpoint

### Additional Information
- `EventReports` endpoint docs -- https://www.misp-project.org/openapi/#tag/Event-Report
- Docs update - https://github.com/MISP/MISP/commit/5b86e5b51f41615f638e115a5d157f6fed93127a
- https://github.com/blackstork-io/fabric/issues/127
- https://github.com/blackstork-io/fabric/issues/126

### PR DESCRIPTION
**Pull request by anasmuhmd**
closes #130

### CODE REVIEW

File: internal/misp/publish_misp_event_reports.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+		datactx["format"] = plugindata.String(params.Format.String())
+		var printer print.Printer
+		switch params.Format {
+		case plugin.OutputFormatMD:
+			printer = mdprint.New()
+		default:
+			return diagnostics.Diag{{
+				Severity: hcl.DiagError,
+				Summary:  "Unsupported format",
+				Detail:   "Only md and html formats are supported",
```
Comment (traut) [inline internal/misp/publish_misp_event_reports.go:105]: incorrect Diag detail message - only MD is supported

File: internal/misp/publish_misp_event_reports.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+		}
+		printer = print.WithLogging(printer, logger, slog.String("format", params.Format.String()))
+		printer = print.WithTracing(printer, tracer, attribute.String("format", params.Format.String()))
+
+		buff := bytes.NewBuffer(nil)
+		err := printer.Print(ctx, buff, document)
+		if err != nil {
+			return diagnostics.Diag{{
+				Severity: hcl.DiagError,
+				Summary:  "Failed to write to a file",
```
Comment (traut) [inline internal/misp/publish_misp_event_reports.go:116]: Incorrect Diag summary

**Review by dobarx — APPROVED** [review]
Looks good! Just few minor issues with messages in diagnostics that you @traut  mentioned.

File: internal/misp/publish_misp_event_reports.go (hunk 2 above)
Comment (anasmuhmd) [reply internal/misp/publish_misp_event_reports.go:116]: @traut thanks, here we're writing the content to a buffer. Does "Failed to render the report" sound good to you?

Comment (traut) [reply internal/misp/publish_misp_event_reports.go:116]: yeah, makes sense. Thanks!

**Comment by traut** [comment]
thanks! The last thing is there is some stale generated code - [the CI check is failing](https://github.com/blackstork-io/fabric/actions/runs/12494885337/job/34865033440?pr=269). You can regenerate the code by running `just generate`

**Pull request merged**